drgn/helpers/linux/net.py

```diff
@@ -45,3 +44,33 @@ def sk_nulls_for_each(head: Object) -> Iterator[Object]:
         "struct sock", head, "__sk_common.skc_nulls_node"
     ):
         yield sk
+
+
+_NETDEV_HASHBITS = 8
+_NETDEV_HASHENTRIES = 1 << _NETDEV_HASHBITS
+
+
+def netdev_get_by_index(
+    prog_or_net: Union[Program, Object], ifindex: IntegerLike
+) -> Object:
+    """
+    Return the corresponding ``struct net_device`` object given an interface
+    index number, or ``NULL`` if not found.
+
+    :param prog_or_net: ``struct net *`` object containing the target device, or
+                        :class:`Program` to use the initial network namespace
+    :param ifindex: Network interface index number of the target device
+    :return: ``struct net_device *``
+    """
+    if isinstance(prog_or_net, Program):
+        prog_or_net = prog_or_net["init_net"]
+
+    head = prog_or_net.dev_index_head[
+        operator.index(ifindex) & (_NETDEV_HASHENTRIES - 1)
+    ]
```

Comment:
I don't think we need to call `operator.index()` here; a `drgn.Object` can be indexed by another `drgn.Object`. But, we probably want to read the ifindex once (otherwise, if it is a reference, we will read it from `/proc/kcore` or the core dump on every comparison):

```suggestion
    if isinstance(ifindex, Object):
        ifindex = ifindex.read_()

    head = prog_or_net.dev_index_head[ifindex & (_NETDEV_HASHENTRIES - 1)]
```

Reply:
Ah, I see; I thought `&` doesn't work on `IntegerLike`.